tests/test_range.py

```diff
@@ -646,6 +646,52 @@ def test_in(self):
                           datetime(2017, 6, 16, 19, 38, 7),
                           datetime(2017, 6, 16, 19, 38, 8))
 
+    def test_since(self):
+        now = datetime.now()
+        self.assert_range('since last may', datetime(2017, 5, 1), now)
+        self.assert_range('since last Friday', datetime(2017, 6, 9), now)
+        self.assert_range('since last Saturday', datetime(2017, 6, 10), now)
+        self.assert_range('since last Thursday', datetime(2017, 6, 15), now)
+        self.assert_range('since last year', datetime(2016, 1, 1), now)
+        self.assert_range('since last month', datetime(2017, 5, 1), now)
+        self.assert_range('since last week', datetime(2017, 6, 5), now)
```

Comment:
datetime(2017, 6, 4), now)

Reply:
Will change week start in a separate PR.